src/metadata/MusicGameMetadataRenderer.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IMetadataRenderer} from "../interfaces/IMetadataRenderer.sol";
+import {IERC721Drop} from "../interfaces/IERC721Drop.sol";
+import {IERC721MetadataUpgradeable} from "@openzeppelin/contracts-upgradeable/interfaces/IERC721MetadataUpgradeable.sol";
+import {IERC2981Upgradeable} from "@openzeppelin/contracts-upgradeable/interfaces/IERC2981Upgradeable.sol";
+import {NFTMetadataRenderer} from "../utils/NFTMetadataRenderer.sol";
+import {MetadataRenderAdminCheck} from "./MetadataRenderAdminCheck.sol";
+
+interface DropConfigGetter {
+    function config()
+        external
+        view
+        returns (IERC721Drop.Configuration memory config);
+}
+
+/// @notice MusicGameMetadataRenderer for editions support
+contract MusicGameMetadataRenderer is
+    IMetadataRenderer,
+    MetadataRenderAdminCheck
+{
+    /// @notice Storage for token edition information
+    struct TokenEditionInfo {
+        string description;
+        string imageURI;
+        string animationURI;
+    }
+
+    /// @notice Event for updated Media URIs
+    event MediaURIsUpdated(
+        address indexed target,
+        address sender,
+        string imageURI,
+        string animationURI
+    );
+
+    /// @notice Event for a new edition initialized
+    /// @dev admin function indexer feedback
+    event NewMusicInitialized(
+        address indexed target,
+        uint256 tokenId,
+        string description,
+        string imageURI,
+        string animationURI
+    );
+
+    /// @notice Description updated for this edition
+    /// @dev admin function indexer feedback
+    event DescriptionUpdated(
+        address indexed target,
+        address sender,
+        string newDescription
+    );
+
+    /// @notice Token information mapping storage
+    mapping(address => mapping(uint256 => TokenEditionInfo))
+        internal _tokenInfos;
+
+    function tokenInfos(address target, uint256 tokenId)
+        public
+        view
+        returns (TokenEditionInfo memory)
+    {
+        return _tokenInfos[target][tokenId];
+    }
+
+    /// @notice Update media URIs
+    /// @param target target for contract to update metadata for
+    /// @param tokenId target token to update metadata for
+    /// @param imageURI new image uri address
+    /// @param animationURI new animation uri address
+    function updateMediaURIs(
+        address target,
+        uint256 tokenId,
+        string memory imageURI,
+        string memory animationURI
+    ) external requireSenderAdmin(target) {
+        _tokenInfos[target][tokenId].imageURI = imageURI;
+        _tokenInfos[target][tokenId].animationURI = animationURI;
+        emit MediaURIsUpdated({
+            target: target,
+            sender: msg.sender,
+            imageURI: imageURI,
+            animationURI: animationURI
+        });
+    }
```

Comment:
is there a case where you would want to update the animationURI or the imageURI alone?

Reply:
possibly, for now, it's not part of our UX.

However, if we need it, we could always just call this fx setting the values we don't want to change as the same value. ✏️ 

If you think this is important, let's chat about the use-case in our UX and how we plan on using it. 🙏